include/tachimawari/control/controller/packet/protocol_1/status/bulk_read_data.hpp

```diff
@@ -36,22 +36,17 @@ namespace tachimawari::control::protocol_1
 class BulkReadData : public Packet
 {
 public:
-  enum
-  {
-    MAX_LENGTH = 255
-  };
+  enum { MAX_LENGTH = 255 };
 
   static void insert_all(
     std::shared_ptr<std::map<uint8_t, BulkReadData>> bulk_data,
     const BulkReadPacket & bulk_read_packet);
 
-  static int validate(
-    std::shared_ptr<std::vector<uint8_t>> rxpacket,
-    int packet_length);
+  static int validate(std::shared_ptr<std::vector<uint8_t>> rxpacket, int packet_length);
 
   static int update_all(
-    std::shared_ptr<std::map<uint8_t, BulkReadData>> bulk_data,
-    std::vector<uint8_t> rxpacket, int packet_length, int data_number);
+    std::shared_ptr<std::map<uint8_t, BulkReadData>> bulk_data, std::vector<uint8_t> rxpacket,
+    int packet_length, int data_number);
```

Comment:
Prioritize `std::unordered_map` over `std::map` if the order of the items does not matter. Additionally, use constant references for most STL containers whenever possible:

```suggestion
    const std::shared_ptr<std::unordered_map<uint8_t, BulkReadData>> & bulk_data, const std::vector<uint8_t> & rxpacket,
    int packet_length, int data_number);
```

Reply:
I changed the map type but not with the const ref implementation at this parameter because there are some mutations inside.

Reply:
Ooh, I see. If that's the case, I recommend using a non-constant reference instead. Also, please change the `rxpacket` to use a const reference as well.

```suggestion
    std::shared_ptr<std::map<uint8_t, BulkReadData>> & bulk_data, const std::vector<uint8_t> & rxpacket,
    int packet_length, int data_number);
```

Reply:
okay for non-const ref.